File: Api/Models/Recipe.cs
namespace RecipeRabbit.Models;

public class Recipe
{
  public int Id { get; set; }
  public string? Name { get; set; }
  public string? Author { get; set; }
  public string? Source { get; set; }
  public string? Servings { get; set; }
  public DateTime DateAddedUtc { get; set; }
  public List<Step> Steps { get; set; } = new();
  public List<RecipeIngredient> RecipeIngredients { get; set; } = new();
}

File: Client/src/models/recipe.ts
import { Ingredient } from "./ingredient"
import { Step } from "./step"

// Data transfer
export interface RecipeSummaryData {
  name: string
  servings?: string
  author?: string
  source?: string
  prepTime?: string
  cookTime?: string
}

export interface RecipeData extends RecipeSummaryData {
  steps: Step[]
  ingredients: Ingredient[]
}

// Complete models
export interface RecipeSummary extends RecipeSummaryData {
  id: number
  dateAddedUtc: string
}

export interface Recipe extends RecipeSummary {
  steps: Step[]
  ingredients: Ingredient[]
}

